@navikt/ds-react/copy-to-clipboard/src/CopyToClipboard.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from "react";
+import { forwardRef, HTMLAttributes, useEffect, useRef, useState } from "react";
+import { Files } from "@navikt/ds-icons";
+import "@navikt/ds-css/baseline/utility.css";
+import "./style.css";
+
+export interface CopyToClipboardProps
+  extends HTMLAttributes<HTMLButtonElement> {
+  /** Verdi som skal kopieres */
+  value: string;
+  /** Beskrivelse av verdien, feks "personnummer", "navn", "epost" etc. */
+  label: string;
+}
+
+const CopyToClipboard = forwardRef<HTMLButtonElement, CopyToClipboardProps>(
+  (props, ref) => {
+    const { value, label, className, ...rest } = props;
+    const textAreaRef = useRef<HTMLTextAreaElement>(null);
+    const [showSuccess, setShowSuccess] = useState(false);
+
+    useEffect(() => {
+      const timeout = setTimeout(() => setShowSuccess(false), 3000);
+      return () => clearTimeout(timeout);
+    }, [showSuccess]);
+
+    const copyToClipboard = (event: React.MouseEvent) => {
+      event.stopPropagation();
+      textAreaRef.current?.select();
+      document.execCommand("copy");
+      setShowSuccess(true);
+    };
+
+    const title = `Kopier ${label} (${value})`;
+
+    return (
+      <span className={`${className} navds-copyToClipboard-wrapper`}>
+        <button
+          ref={ref}
+          title={title}
+          {...rest}
+          className="navds-copyToClipboard-button"
+          onClick={copyToClipboard}
+        >
+          <Files />
+          <span className="sr-only">{title}</span>
+        </button>
+        {showSuccess && (
+          <span role="alert" className="navds-copyToClipboard-success">
```

Comment:
Her har jeg også spørsmål om UU. Blir denne pop-upen håndtert riktig av skjermlesere? Hva er en god timing for midlertidige beskjeder? Jeg tar det opp i neste møtet med UU.

Reply:
Testet den med skjermleser og syntes det funka ok, men sikkert lurt å ta det med UU ja :)